packages/mwp-api-proxy-plugin/src/util/send.js

```diff
@@ -82,54 +82,79 @@ function makeMockResponse(requestOpts, response = MOCK_RESPONSE_OK) {
 }
 
 /**
- * In order to receive cookies from `externalRequest` requests, this function
- * provides a cookie jar that is specific to the request.
+ * Function to build argument for `request` for API requests to DIY edge endpoints,
+ * i.e. endpoints served on a domain other than api.meetup.com
  *
- * The `requestUrl` is used to determine whether a cookie jar is needed
- *
- * https://github.com/request/request#examples
- *
- * @param {String} requestUrl the URL that will be used in the external request
- * @return {Object} a cookie jar compatible with the npm request `jar` API
+ * Currently, cookies cannot be set by DIY edge endpoints. (e.g. using the
+ * `jar` interface in `request`)
  */
-export const createCookieJar = requestUrl => {
-	const parsedUrl = url.parse(requestUrl);
-	if (parsedUrl.pathname === '/sessions') {
-		return externalRequest.jar(); // create request/url-specific cookie jar
-	}
-	return null;
+const buildDIYEdgeRequestArgs = externalRequestOpts => {
+	// make a copy of the 'global' headers set for all queries.
+	const headers = { ...externalRequestOpts.headers };
+	// all DIY edges support JSON bodies for all methods
+	headers['content-type'] = 'application/json';
+	// strip clicktracking cookie 'click-track'
+	headers.cookie = headers.cookie.replace(/\s+?click-track=[^;]+/, '');
+
+	return query => {
+		let body;
+
+		// create properly-escaped URL from endpoint
+		const url = new URL(query.endpoint);
+
+		switch (externalRequestOpts.method) {
+			case 'patch':
+			case 'put':
+			case 'post':
+				if (externalRequestOpts.formData) {
+					break;
+				}
+				// assume all DIY edge endpoints handle JSON body encoding
+				body = JSON.stringify(query.params);
+				break;
+			case 'delete':
+			case 'get':
+			default: {
+				// copy query object params into `url.searchParams`
+				for (const p of new URLSearchParams(query.params)) {
```

Comment:
Does having the `new` operator inside the for-of statement cause it to create a new object on every iteration of the loop? In examples, I've normally seen:

`const param = new URLSearchParams...; for (const p of param) ...`

Reply:
it's not evaluating the second part of `of` every loop, no - same way it doesn't re-evaluate `param` in your example. Examples break it up into two lines so that you can see/re-use the `URLSearchParams` object after the loop, but it's intended to be ephemeral here